square.go

```diff
@@ -32,15 +32,24 @@ func Build(txs [][]byte, maxSquareSize, subtreeRootThreshold int) (Square, [][]b
 		if isBlobTx {
 			if builder.AppendBlobTx(blobTx) {
 				blobTxs = append(blobTxs, txBytes)
+			} else {
+				// The blobTx can not fit so exit the txs loop and return the square.
+				break
 			}
 		} else {
 			if builder.AppendTx(txBytes) {
 				normalTxs = append(normalTxs, txBytes)
+			} else {
+				// The normal tx can not fit so exit the txs loop and return the square.
+				break
```

Comment:
for posterity, leaving some thoughts here

this has the tradeoff of not including txs that otherwise would fit.

the max wasted space is that of the max transaction size, which is currently 2MiB

Reply:
irl, it would likely be less than the max size most of the time.